docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,218 @@
+```
+
+5.  Edit grub to enable intel-iommu:
+
+```
+edit /etc/default/grub and add intel_iommu=on to cmdline:
+- GRUB_CMDLINE_LINUX=""
++ GRUB_CMDLINE_LINUX="intel_iommu=on"
+sudo update-grub
+```
+
+6.  Reboot and verify
+Host system should be ready now -- reboot and verify that expected cmdline and
+kernel version is booted (look at /proc/cmdline and /proc/version)
+
+```
+sudo reboot
+```
+
+## Setting up SRIO-V Device
+
+All of the prior sections are needed once to prepare the SRIOV host systems.
+The following will be needed per boot in order to facilitate setting up a
+physical device's virtual functions.
+
+For SRIOV, a physical device can create up to `sriov_totalvfs` virtual 
+functions (VFs).  Once created, you cannot grow or shrink the number of VFs 
+without first setting it back to zero.  Based on this, it is expected that you
+should set the number of VFs from a physical device  just once.
+
+1. Add vfio-pci device driver
+
+vfio-pci is a driver which is used to reserve a VF PCI device.  Add it:
+```
+sudo modprobe vfio-pci
+```
+2. Find our NICs of interest
+
+Find PCI details for the NICs in question:
+```
+sriov@sriov-1:/sys/bus/pci$ lspci | grep Ethernet
+00:19.0 Ethernet controller: Intel Corporation Ethernet Connection I217-LM (rev 04)
+01:00.0 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+01:00.1 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+```
+In our case, both 01:00.0 and 01:00.1 are the two ports on our x540-AT2 card 
+that we'll use.  You can use lshw to get further details on the controller 
+and verify it indeed supports SRIO-V.
+
+3.  Check how many VFs we can create:
+```
+$ sriov@sriov-1:~$ cat /sys/bus/pci/devices/0000\:01\:00.0/sriov_totalvfs
+63
+$ sriov@sriov-1:~$ cat /sys/bus/pci/devices/0000\:01\:00.1/sriov_totalvfs
+63
+```
+4.  Create VFs:
+
+Create virtual functions by editing sriov_numvfs. In our example, let's 
+just create one per physical device.  Note, this eliminates the usefulness 
+of SRIOV, and is just done for simplicity in this example so you needn't
+look at 128 virtual devices.
+
+```
+root@sriov-1:/home/sriov# echo 1 > /sys/bus/pci/devices/0000\:01\:00.0/sriov_numvfs
+root@sriov-1:/home/sriov# echo 1 > /sys/bus/pci/devices/0000\:01\:00.1/sriov_numvfs
+```
+
+5.  Verify that these were added to the host:
+
+```
+root@sriov-1:/home/sriov# lspci | grep Ethernet | grep Virtual
+02:10.0 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+02:10.1 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+```
+6.  Assign MAC address to each VF
+
+Depending on the NIC being used, you may need to explicitly set the MAC
+address for the Virtual Function device in order to guarantee that the 
+address is consistent on the host and when passed through to the guest.
+For example,
+
+```
+ip link set <pf> vf <vfidx> mac <my made up mac address>
```

Comment:
Might be better to change "my made up mac address" to simply "fake MAC address".